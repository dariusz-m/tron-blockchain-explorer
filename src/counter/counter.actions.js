export const CounterActionType = {
    INCREMENT: 'INCREMENT',
    DECREMENT: 'DECREMENT'
};

export const increment = () => {
    return {
        type: CounterActionType.INCREMENT
    };
};

export const decrement = () => {
    return {
        type: CounterActionType.DECREMENT
    };
};
