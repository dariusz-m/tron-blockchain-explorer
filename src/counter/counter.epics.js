import "rxjs/add/operator/do";
import "rxjs/add/operator/mapTo";
import "rxjs/add/operator/mergeMap";

import { fromPromise } from 'rxjs/observable/fromPromise';

import {TronBlockchainClient} from '../services/tron-api';
import {CounterActionType} from './counter.actions';

export const increment = action$ =>
    action$.ofType(CounterActionType.INCREMENT)
        .mergeMap(() => fromPromise(TronBlockchainClient.getLatestBlock()))
        .mergeMap((block) => fromPromise(TronBlockchainClient.getBlockByNum(block.number)))
        .do((data) => {
            console.log(data);
        })
        .mapTo({type: "test"});
