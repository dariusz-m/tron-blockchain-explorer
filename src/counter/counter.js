import React from 'react';
import PropTypes from 'prop-types';
import {connect} from "react-redux";
import {decrement, increment} from "./counter.actions";
import {bindActionCreators} from "redux";

export class Counter extends React.Component {

    constructor() {
        super();
    }

    render() {
        return (
            <div>
                <h1>Test</h1>
                <h2>Counter: {this.props.counter}</h2>
                <button onClick={this.props.actions.increment}>INC</button>
                <button onClick={this.props.actions.decrement}>DEC</button>
            </div>
        );
    }
}

Counter.propTypes = {
    counter: PropTypes.number.isRequired,
    actions: PropTypes.shape({
        increment: PropTypes.func.isRequired,
        decrement: PropTypes.func.isRequired,
    }),
};

function mapStateToProps(state) {
    return {
        counter: state.counter
    };
}

function mapDispatchToProps(dispatch) {
    return {
        actions: bindActionCreators({
            increment: increment,
            decrement: decrement
        }, dispatch)
    };
}

export default connect(mapStateToProps, mapDispatchToProps)(Counter);
