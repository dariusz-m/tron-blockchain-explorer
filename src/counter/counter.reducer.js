import initialState from '../state/initialState';
import {CounterActionType} from './counter.actions';

export default (state = initialState.counter, action) => {
    switch (action.type) {
        case CounterActionType.INCREMENT:
            return state + 1;
        case CounterActionType.DECREMENT:
            return state - 1;
        default:
            return state;
    }
};
