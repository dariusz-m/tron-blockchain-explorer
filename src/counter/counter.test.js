import React from 'react';
import renderer from 'react-test-renderer';
import {Counter} from "./counter";

describe('Counter tests', () => {
    it('renders correctly', () => {
        const actions = {
            increment: () => {},
            decrement: () => {},
        };
        const tree = renderer.create(<Counter actions={actions} counter={1}/>).toJSON();
        expect(tree).toMatchSnapshot();
    });
});

