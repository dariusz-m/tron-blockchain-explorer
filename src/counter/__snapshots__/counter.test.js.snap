// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter tests renders correctly 1`] = `
<div>
  <h2>
    Counter: 
    1
  </h2>
  <button
    onClick={[Function]}
  >
    INC
  </button>
  <button
    onClick={[Function]}
  >
    DEC
  </button>
</div>
`;
